core/i2c/i2cdriver.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * @file i2cdriver.h
+ * @author Michal Mankowski (m.mankowski2004@gmail.com)
+ * @brief This file defines I2C driver
+ * @version 0.1
+ * @date 2024-01-14
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#ifndef CORE_I2C_I2CDRIVER_H_
+#define CORE_I2C_I2CDRIVER_H_
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <linux/i2c-dev.h>
+#include <linux/types.h>
+#include <linux/i2c.h>
+#include <unistd.h>
+#include <vector>
+#include "core/common/error_code.h"
+#include "core/results/result.h"
+#include <string>
+namespace simba {
+namespace core {
+class I2C{
+ public:
+    ErrorCode init(const std::string& path);
+    ErrorCode read(const uint8_t address, const uint8_t reg, std::vector<uint8_t>& vect);
```

Comment:
Troche slabo jest przez referencje bo cpplint bedzie wywalal sie + mozesz uzyc core::Result<td::vector<uint8_t>> jako zwracanie i jak bedzie blad to zwracamy pusty obiekt
```suggestion
    ErrorCode read(const uint8_t address, const uint8_t reg, std::week_ptr<std::vector<uint8_t>> vect);
```